frontend/src/components/SocialButton.tsx

```diff
@@ -0,0 +1,51 @@
+
+import React from 'react';
+import {Text, TouchableOpacity, View, StyleSheet} from 'react-native';
+
+import FontAwesome from 'react-native-vector-icons/FontAwesome';
+
+const SocialButton = ({buttonTitle, btnType, color, backgroundColor, ...rest}) => {
+  let bgColor = backgroundColor;
+  return (
+    <TouchableOpacity
+      style={[styles.buttonContainer, {backgroundColor: bgColor}]}
+      {...rest}>
```

Comment:
This ...rest pattern is a bit weird, not really seen it around - what happens if you don't spread the rest of the props?